pennylane_qiskit/qiskit_device.py

```diff
@@ -119,26 +99,36 @@ class QiskitDevice(Device, abc.ABC):
     observables = {"PauliX", "PauliY", "PauliZ", "Identity", "Hadamard", "Hermitian"}
 
     hw_analytic_warning_message = (
-        "The analytic calculation of expectations and variances "
-        "is only supported on statevector backends, not on the {}. "
-        "The obtained result is based on sampling."
+        "The analytic calculation of expectations, variances and "
+        "probabilities is only supported on statevector backends, not on the {}. "
+        "Such statistics obtained from this device are estimates based on samples."
     )
 
     _eigs = {}
 
     def __init__(self, wires, provider, backend, shots=1024, **kwargs):
         super().__init__(wires=wires, shots=shots)
 
+        # Keep track if the user specified analytic to be True
+        analytic_was_specified = "analytic" in kwargs and kwargs["analytic"]
         self.analytic = kwargs.pop("analytic", False)
```

Comment:
Yes, if it was not set explicitly by the user (that is not part of `kwargs`). The `pop` is needed so that this will not be carried on in the `run_args` later.

One thing to keep in mind is that with the `super().__init__(wires=wires, shots=shots)` call we are already setting `analytic=True` by default from `QubitDevice.__init__`. More details on this topic are in this comment: https://github.com/XanaduAI/pennylane-qiskit/pull/83#discussion_r413243678

Reply:
Thanks!